x-pack/test/api_integration/apis/endpoint/resolver.ts

```diff
@@ -79,14 +60,59 @@ const verifyAncestry = (ancestors: LifecycleNode[], tree: Tree, verifyLastParent
   expect(Object.keys(groupedAncestors).length).to.eql(ancestors.length);
   // make sure there aren't any nodes with the same parent entity_id
   expect(Object.keys(groupedAncestorsParent).length).to.eql(ancestors.length);
-  ancestors.forEach((node) => {
+
+  // make sure each of the ancestors' lifecycle events are in the generated tree
+  for (const node of ancestors) {
+    expectLifecycleNodeInMap(node, tree.ancestry);
+  }
+
+  // start at the origin which is always the first element of the array and make sure we have a connection
+  // using parent id between each of the nodes
+  let foundParents = 0;
+  let node = ancestors[0];
+  for (let i = 0; i < ancestors.length; i++) {
     const parentID = parentEntityId(node.lifecycle[0]);
-    // the last node generated will have `undefined` as the parent entity_id
-    if (parentID !== undefined && verifyLastParent) {
-      expect(groupedAncestors[parentID]).to.be.ok();
+    if (parentID !== undefined) {
+      const nextNode = groupedAncestors[parentID];
+      if (!nextNode) {
+        break;
+      }
+      // the grouped nodes should only have a single entry since each entity is unique
+      node = nextNode[0];
     }
-    expectLifecycleNodeInMap(node, tree.ancestry);
-  });
+    foundParents++;
+  }
+
+  if (verifyLastParent) {
+    expect(foundParents).to.eql(ancestors.length);
+  } else {
+    // if we only retrieved a portion of all the ancestors then the most distant grandparent's parent will not necessarily
+    // be in the results
+    expect(foundParents).to.eql(ancestors.length - 1);
+  }
+};
+
+/**
+ * Retrieves the most distant ancestor in the given array.
+ *
+ * @param ancestors an array of ancestor nodes
+ */
+const retrieveDistantAncestor = (ancestors: LifecycleNode[]) => {
```

Comment:
The way the ancestry array is implemented now with a max of 2 if I read that correctly, this will only currently retrieve the grandparent?

Reply:
Ignore me, saw it was configurable in the test below.